packages/api/cms-api/src/generator/generate-crud-input.ts

```diff
@@ -50,32 +56,49 @@ export async function generateCrudInput(
+            const defaultValue = prop.nullable && (initializer == "undefined" || initializer == "null") ? "null" : initializer;
+            const fieldOptions = tsCodeRecordToString({ nullable: prop.nullable ? "true" : undefined, defaultValue });
             const enumName = findEnumName(prop.name, metadata);
             const importPath = findEnumImportPath(enumName, generatorOptions, metadata);
             importsOut += `import { ${enumName} } from "${importPath}";`;
             decorators.push(`@IsEnum(${enumName})`);
-            decorators.push(`@Field(() => ${enumName}${prop.nullable ? ", { nullable: true }" : ""})`);
+            decorators.push(`@Field(() => ${enumName}, ${fieldOptions})`);
             type = enumName;
         } else if (prop.type === "string") {
+            const initializer = morphTsProperty(prop.name, metadata).getInitializer()?.getText();
+            const defaultValue = prop.nullable && (initializer == "undefined" || initializer == "null") ? "null" : initializer;
+            const fieldOptions = tsCodeRecordToString({ nullable: prop.nullable ? "true" : undefined, defaultValue });
             decorators.push("@IsString()");
             if (prop.name.startsWith("scope_")) {
                 continue;
             } else if (prop.name === "slug") {
                 //TODO find a non-magic solution
                 decorators.push("@IsSlug()");
             }
-            decorators.push(`@Field(${prop.nullable ? "{ nullable: true }" : ""})`);
+            decorators.push(`@Field(${fieldOptions})`);
         } else if (prop.type === "DecimalType" || prop.type === "number") {
+            const initializer = morphTsProperty(prop.name, metadata).getInitializer()?.getText();
+            const defaultValue = prop.nullable && (initializer == "undefined" || initializer == "null") ? "null" : initializer;
+            const fieldOptions = tsCodeRecordToString({ nullable: prop.nullable ? "true" : undefined, defaultValue });
             decorators.push("@IsNumber()");
-            decorators.push(`@Field(${prop.nullable ? "{ nullable: true }" : ""})`);
+            decorators.push(`@Field(${fieldOptions})`);
             type = "number";
         } else if (prop.type === "DateType" || prop.type === "Date") {
             decorators.push("@IsDate()");
             decorators.push(`@Field(${prop.nullable ? "{ nullable: true }" : ""})`);
             type = "Date";
         } else if (prop.type === "BooleanType" || prop.type === "boolean") {
+            const initializer = morphTsProperty(prop.name, metadata).getInitializer()?.getText();
+            const defaultValue =
+                prop.nullable && (initializer == "undefined" || initializer == "null")
+                    ? "null"
+                    : initializer == "true" || initializer == "false"
+                    ? initializer
+                    : undefined;
```

Comment:
Why can't this be `const defaultValue = prop.nullable && (initializer == "undefined" || initializer == "null") ? "null" : initializer;` as in the other cases?

Reply:
My initial idea was to support only literals but I gave up on that for the other cases. Could be changed here for consistency.

Reply:
Yes, please change it 🙂